lib/index.js

```diff
@@ -4,165 +4,88 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import {syncPR} from './sync/index.js';
+import {isSyncStatusContext} from './sync/commit-status.js';
 import {
-  closePR,
-  createParentPR,
-  createPlaceholderStatus,
-  genQueueKey,
-  getPartnerPR,
-  getPRFromChecksPayload,
-  getPRFromPushPayload,
-  getPRFromStatusPayload,
-  mergePR,
-  pushPayloadHasCopyableCommits,
-  reopenPR,
-  syncChildPR,
-  syncChildRepos,
-  syncParentPR,
-  syncPRStatus,
-} from './pr-utils.js';
-import {hasChildren, hasParent, hasRelationship} from './relationship-utils.js';
-import {queue} from './utils.js';
-
-const STATUS_CONTEXTS = {
-  child: 'child-monorepo/ci',
-  parent: 'parent-monorepo/ci',
-};
-
-export default function(app) {
-  app.on('check_suite.requested', onCheckSuiteRequested);
-  app.on(
-    ['pull_request.closed', 'pull_request.reopened'],
-    onPullRequestStateChange,
-  );
-  app.on('pull_request.closed', onPullRequestMerge);
-  app.on('pull_request.opened', onPullRequestOpen);
-  app.on('push', onPush);
-  app.on('status', onCommitStatusChange);
-
-  // add placeholder parent/child PR status to prevent merging during
-  // sync delay
-  async function onCheckSuiteRequested({payload}) {
-    const repoName = payload.repository.full_name;
-
-    if (!hasRelationship(repoName)) {
-      return;
-    }
-
-    const PR = await getPRFromChecksPayload(payload);
-
-    if (PR) {
-      const statusContext =
-        STATUS_CONTEXTS[hasParent(repoName) ? 'parent' : 'child'];
-
-      await queue(genQueueKey(PR, 'status'), () =>
-        createPlaceholderStatus(PR, statusContext),
-      );
-    }
-  }
-
-  // create parent PR for child PR
-  async function onPullRequestOpen({payload}) {
-    if (!hasParent(payload.repository.full_name)) {
-      return;
-    }
-
-    // TODO: should this assume parent PR doesn't exist?
-    await createParentPR(payload.pull_request);
-  }
-
-  // when parent or child PR is closed/reopened, close/reopen the other
-  async function onPullRequestStateChange({payload}) {
-    if (
-      payload.pull_request.merged ||
-      !hasRelationship(payload.repository.full_name)
-    ) {
-      return;
-    }
-
-    const includeClosed = payload.action === 'reopened';
-    const partnerPR = await getPartnerPR(payload.pull_request, includeClosed);
-
-    if (partnerPR) {
-      await queue(genQueueKey(partnerPR, 'state-change'), () => {
-        switch (payload.action) {
-          case 'closed':
-            return closePR(partnerPR);
-          case 'reopened':
-            return reopenPR(partnerPR);
-        }
+  getGithubId,
+  getProbotRepoNames,
+  request,
+  saveAppGithubClient,
+} from './github.js';
+import {hasRelationship} from './relationships.js';
+import {throttleWebhook} from './utils.js';
+
+async function init(repoNames) {
+  for (const repoName of repoNames) {
```

Comment:
Does this need to execute in serial? if not we could speed this up by running in parallel.
```
Promise.all(repoNames.filter(r => hasRelationship(r)).map(r => ... etc)))
```

Reply:
In a general sense it could run in parallel. However with the way we're using it (deployed as 2 separate apps on each org) I think I'm actually gonna need to modify this to avoid the 2 deployments doing double work (since `syncPR` actually syncs both repos involved in the pr). 

Longer term I think I'm gonna see if I can fix the issues we had with just using one deployment for both orgs

Reply:
Nevermind, actually fixing this issue in https://github.com/fusionjs/probot-app-workflow/pull/147